consensus/istanbul/backend/api.go

```diff
@@ -89,6 +90,16 @@ func (api *API) GetValidators(number *rpc.BlockNumber) ([]common.Address, error)
 	return istanbul.MapValidatorsToAddresses(validators), nil
 }
 
+// GetValidatorsPublicKeys retrieves the list of validators public keys that must sign a given block.
+func (api *API) GetValidatorsPublicKeys(number *rpc.BlockNumber) ([]blscrypto.SerializedPublicKey, error) {
```

Comment:
how is `blscrypto.SerializedPublicKey` encoded to json? does it have `Marshall/Unmarshall` methods already defined?

Reply:
Hmm, good point. I've seen it was passed as byte arrays between Rust and Go, and in JSON it's serialized as an array of integers.

Reply:
Normally we use hex strings. you can use the `hexutil.Bytes` check https://github.com/celo-org/celo-blockchain/blob/d42b997916c74cc17a623e341618cd8190b27d9a/internal/ethapi/api.go#L691

Reply:
Sounds good :)
How does it happen automatically for `common.Address`? I see methods returning it directly.

Reply:
Because of...
```
// MarshalText returns the hex representation of a.
func (a Address) MarshalText() ([]byte, error) {
	return hexutil.Bytes(a[:]).MarshalText()
}

// UnmarshalText parses a hash in hex syntax.
func (a *Address) UnmarshalText(input []byte) error {
	return hexutil.UnmarshalFixedText("Address", input, a[:])
}

// UnmarshalJSON parses a hash in hex syntax.
func (a *Address) UnmarshalJSON(input []byte) error {
	return hexutil.UnmarshalFixedJSON(addressT, input, a[:])
}

```

Reply:
Same happens for Hash, definitely the way to do it.

Check https://github.com/celo-org/celo-blockchain/blob/0ec5ab417559ac3399a73f76219fce043dfc0ef9/common/types.go#L253

Reply:
Thanks! Done :)